easy-ui-react/src/ProductLayout/ProductLayoutHeader.tsx

```diff
@@ -0,0 +1,159 @@
+import { CollectionChildren } from "@react-types/shared";
+import React, { Key, ReactNode } from "react";
+import { Button } from "../Button";
+import { Icon } from "../Icon";
+import { Text } from "../Text";
+import { UnstyledButton } from "../UnstyledButton";
+import { classNames } from "../utilities/css";
+import { HelpMenu } from "./HelpMenu";
+import { ScreenSizeSwitcher } from "./ScreenSizeSwitcher";
+import { useProductLayout } from "./context";
+
+import styles from "./ProductLayoutHeader.module.scss";
+
+export type ProductLayoutHeaderProps = {
+  /**
+   * List of help menu items to render. Should be an array of `<Menu.Item />`s.
+   */
+  helpMenuItems: CollectionChildren<object>;
+
+  /**
+   * Handler that is called when a help menu item is selected.
+   */
+  onHelpMenuAction?: (key: Key) => void;
+
+  /**
+   * Primary call to action for the header.
+   */
+  primaryAction?: ProductLayoutHeaderActionProps;
+
+  /**
+   * Logo to render in the header on small screens.
+   */
+  renderSmallScreenLogo: () => ReactNode;
+
+  /**
+   * Secondary call to action for the header.
+   */
+  secondaryAction?: ProductLayoutHeaderActionProps;
+
+  /**
+   * Page title.
+   */
+  title: ReactNode;
+};
+
+export type ProductLayoutHeaderActionProps = {
+  /**
+   * Content text to render for the header action.
+   */
+  content: string;
+
+  /**
+   * Action to invoke for the header action.
+   */
+  onAction: () => void;
+
+  /**
+   * Whether or not the header action is disabled.
+   */
+  isDisabled?: boolean;
+};
+
+export function ProductLayoutHeader(props: ProductLayoutHeaderProps) {
+  return (
+    <header>
+      <ScreenSizeSwitcher
+        renderOnLargeScreen={() => <LargeScreenHeader {...props} />}
+        renderOnSmallScreen={() => <SmallScreenHeader {...props} />}
+      />
+    </header>
+  );
+}
+
+function SmallScreenHeader(props: ProductLayoutHeaderProps) {
+  const {
+    helpMenuItems,
+    onHelpMenuAction,
+    primaryAction,
+    renderSmallScreenLogo,
+    title,
+  } = props;
+  const { sidebarTriggerProps } = useProductLayout();
+  return (
+    <div className={classNames(styles.ProductLayoutHeader, styles.smallScreen)}>
+      <div className={styles.smallScreenTopBar}>
+        <div className={styles.logoMenu}>
+          <UnstyledButton
+            className={styles.logoMenuBtn}
+            {...sidebarTriggerProps}
+          >
+            <Icon symbol={MenuIcon} />
+            <Text visuallyHidden>Menu</Text>
+          </UnstyledButton>
+          {renderSmallScreenLogo()}
+        </div>
+        <div className={styles.actions}>
+          {primaryAction ? (
+            <Button
+              size="sm"
+              onPress={primaryAction.onAction}
+              isDisabled={primaryAction.isDisabled}
+            >
+              {primaryAction.content}
+            </Button>
+          ) : (
+            <HelpMenu items={helpMenuItems} onAction={onHelpMenuAction} />
+          )}
```

Comment:
on small screens only either the primary CTA is shown, otherwise the help menu is